lib/rust/lexer/definition/src/lexer.rs

```diff
@@ -464,9 +722,805 @@ impl EnsoLexer {
+    fn on_escape_invalid<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Text::on_escape_invalid");
+        let offset  = self.offset.consume();
+        let current = self.consume_current();
+        let escape  = Token::TextSegmentEscape(token::EscapeStyle::Invalid,current,offset);
+        self.text_state.append_segment(escape);
+    }
+
+    // === Rule Definitions ===
+
     /// Define the rules for lexing Enso text literals.
-    fn add_text_rules(_lexer:&mut EnsoLexer) {
-        // TODO [AA] Write the lexing rules for text literals.
+    fn add_text_rules(lexer:&mut EnsoLexer) {
+        let format_quote         = c!('\'');
+        let format_block_quote   = l!("'''");
+        let raw_quote            = c!('"');
+        let raw_block_quote      = l!("\"\"\"");
+        let backslash            = c!('\\');
+        let spaces               = EnsoLexer::spaces();
+        let eof                  = Pattern::eof();
+        let interpolate_quote    = c!('`');
+        let lf                   = l!("\n");
+        let crlf                 = l!("\r\n");
+        let format_block_lf      = &format_block_quote >> spaces.opt() >> &lf;
+        let format_block_crlf    = &format_block_quote >> spaces.opt() >> &crlf;
+        let raw_block_lf         = &raw_block_quote >> spaces.opt() >> &lf;
+        let raw_block_crlf       = &raw_block_quote >> spaces.opt() >> &crlf;
+        let format_char          = Pattern::none_of("'`\\\n\r");
+        let format_raw_segment   = format_char.many1();
+        let format_block_segment = Pattern::none_of("\n\r\\`").many1();
+        let raw_block_segment    = Pattern::none_of("\n\r\\");
+        let raw_segment          = Pattern::none_of("\"\n\r\\").many1();
+        let unicode_escape_char  = Pattern::none_of("'\"`\\\n\r{}");
+        let escape_byte          = l!("\\x") >> Pattern::repeat_between(&unicode_escape_char,0,3);
+        let escape_u16           = l!("\\u") >> Pattern::repeat_between(&unicode_escape_char,0,5);
+        let escape_u21           = l!("\\u{") >> &unicode_escape_char.many() >> l!("}");
+        let escape_u32           = l!("\\U") >> Pattern::repeat_between(&unicode_escape_char,0,9);
+        let escape_slash         = l!("\\\\");
+        let escape_invalid       = &backslash >> Pattern::not_symbol(Symbol::eof()).opt();
+        let escape_format_quote  = &backslash >> &format_quote;
+        let escape_raw_quote     = &backslash >> &raw_quote;
+        let escape_backtick      = &backslash >> &interpolate_quote;
+        let invalid_format_quote = &format_block_quote >> format_quote.many1();
+        let invalid_raw_quote    = &raw_block_quote >> raw_quote.many1();
+        let eof_new_line         = &spaces.opt() >> &eof;
+        let empty_line_lf        = &spaces.opt() >> &lf;
+        let empty_line_crlf      = &spaces.opt() >> &crlf;
+        let comment_char         = l!("#").many1();
+        let doc_comment_start    = l!("## ");
```

Comment:
why does it contain a space? So doc comments like `##foo` will not work? The parser should be VERY permissive - even if soemthing doesnt look nice it should be parsed